planner/core/logical_plan_builder.go

```diff
@@ -3237,6 +3236,20 @@ func (b *PlanBuilder) buildMemTable(_ context.Context, dbName model.CIStr, table
 
 // BuildDataSourceFromView is used to build LogicalPlan from view
 func (b *PlanBuilder) BuildDataSourceFromView(ctx context.Context, dbName model.CIStr, tableInfo *model.TableInfo) (LogicalPlan, error) {
+	viewFullName := dbName.L + "." + tableInfo.Name.L
```

Comment:
Can we extract these code block as in individual function like `checkRecrusiveView`?

Reply:
addressed, PTAL